inbox/crispin.py

```diff
@@ -825,19 +826,19 @@ def condstore_supported(self) -> bool:
     def idle_supported(self) -> bool:
         return b"IDLE" in self.conn.capabilities()
 
-    def search_uids(self, criteria: List[str]) -> List[int]:
+    def search_uids(self, criteria: List[str]) -> Iterable[int]:
         """
         Find UIDs in this folder matching the criteria. See
         http://tools.ietf.org/html/rfc3501.html#section-6.4.4 for valid
         criteria.
 
         """
-        return sorted(
```

Comment:
The sorted call here was doing useless extra work (creating a sorted list) and adding memory pressure, because we are converting most of the time to sets in the caller. When we are *not* converting to sets the order does not matter for the caller.